src/components/RoomNameInput/index.js

```diff
@@ -65,6 +71,9 @@ class RoomNameInput extends Component {
                 onSelectionChange={event => this.setSelection(event.nativeEvent.selection)}
                 errorText={this.props.errorText}
                 autoCapitalize="none"
+                onBlur={this.props.onBlur}
```

Comment:
If we apply the `inputID=roomName` directly to the `TextInput` component instead of applying it to the `RoomName` component in `WorkspaceNewRoomPage` would we avoid having to pass these `onBlur` and `autoFocus` methods as props?

Reply:
Correct! But I’m not sure that’s a good practice where the `RoomNameInput` will be exclusively used in Form.js only.

Reply:
@puneetlath I think this won't work because `RoomNameInput` is wrapped in HOC `withLocalize` , and current `Form` doesn't support that

Reply:
Ahh, so this allows us to get around the HOC. I see!

Reply:
Ok so question actually. What do you all think is cleaner: 

1. Keeping the HOC wrapper and passing `inputID` and `onBlur` as props. 
2. Removing the HOC wrapper and passing translated `label` and `placeholder` as props

Reply:
@puneetlath  2nd proposal looks promising to me , `WorkspaceNewRoomPage` is already using `withLocalize` ,
we need to refactor `ReportSettingsPage` to pass newly added props too

Reply:
I was thinking we would refactor `ReportSettingsPage` to use the `Form` component in a separate PR. But maybe it is better to do here given the overlap (we can increase the bounty if we decide to increase the scope). But basically I'm thinking: 

1. Update `WorkspaceNewRoomPage` and `ReportSettingsPage` to use `Form`
2. Update them to pass `label` and `placeholder` to `RoomNameInput`
3. Remove HOC wrapper from `RoomNameInput`
4. Apply `inputID` directly to the `TextInput` in `RoomNameInput`

That feels like it would be in line with how other forms are done as well. @mollfpr @fedirjh thoughts?

Reply:
@puneetlath I think we have limitation here:

1. In the actual implementation the `Form` controls the whole `RoomNameInput` class , and the `RoomNameInput` control `TextInput` , so when data changes inside the `TextInput` then `setModifiedRoomName` will modify `value` and will trigger the `onInputChange` of the Form input . **If we apply** `inputID` directly to the `TextInput` in `RoomNameInput` , the `Form` will control the `TextInput` directly and this will create a conflict (we cannot use `setModifiedRoomName` to change value) , this will require extra change to `Form` or another workaround , I think we got this limitation [here](https://github.com/Expensify/App/issues/13133)
2. `ReportSettingsPage` : `Form` uses `ScrollView` with a submit Button at the bottom , ReportSettingsPage's design is a bit different from regular Forms , is it possible for the `Form` to fit that design ?

My thoughts is that we can have a 3rd solution , We can make changes similar to [**Picker.js**](https://github.com/Expensify/App/blob/9d357e459b1b0114d60f5b6162ea6ee9672356aa/src/components/Picker/index.js) : 
- Removing the HOC wrapper and passing translated `label` and `placeholder` as props 
- Passing `inputID` and `onBlur` as props so Form can handle the `RoomNameInput` and we can control the `value` , I think after all that `RoomNameInput` utility's to allow us to rewrite the `value`.

For `ReportSettingsPage` I have no problem to refactor it here but i think it should be discussed with design team because I think the design will change. @puneetlath @mollfpr  thoughts?

Reply:
Ok got it. Given all that, I think it's best to go forward with what we have here and we can handle `ReportSettingsPage` separately. Thanks for the discussion!